Demo/Demo/FeatureViewControllers/FeatureBaseViewController.swift

```diff
@@ -87,15 +86,37 @@ class FeatureBaseViewController: UIViewController {
                 self.updateTitle("Order ID: \(order.id)")
                 self.orderID = order.id
                 print("✅ fetched orderID: \(order.id) with status: \(order.status)")
+                completion(order.id)
             case .failure(let error):
                 self.updateTitle("Your order has failed, please try again")
                 print("❌ failed to fetch orderID: \(error)")
+                completion(nil)
             }
         }
     }
 
-    func processOrder() {
-        // TODO: Pass in card details once card client is merged and call processOrder from DemoMerchantAPI
+    func processOrder(orderData: OrderData, completion: @escaping () -> Void = {}) {
+        var intent = DemoSettings.intent.rawValue.capitalized
+        intent.removeLast()
```

Comment:
why are we calling `removeLast` on an `Intent` enum? is it just to trim the string? 

i feel like we're saving some lines of code from a `switch` statement here that is costing us clarity, since it's not intuitive that we're treating the intent `String` like and `Array` here by dropping the last character. What do you think?

Reply:
yea we did this to build the title text `Authorizing/Capturing order ...` below.
I agree a switch statement will be clearer here!!! :smile: